libs/ledger/src/execution_manager.cpp

```diff
@@ -588,8 +591,13 @@ void ExecutionManager::MonitorThreadEntrypoint()
           FETCH_LOCK(idle_executors_lock_);
           if (!idle_executors_.empty())
           {
+            Address last_block_miner{};
+            summary_.Apply([&last_block_miner](Summary const &summary) {
```

Comment:
I think it will be better to use rather `SynchronisedState<T>::Get()` method in order to *get* the  out of the thread-safe state - code will be more readable and also computationally cheaper as well, at least for `T = Summary` (no need to contruct lambda with *capture*).
```cpp
Address last_block_miner {summary_.Get().last_block_miner};
```

Reply:
Minor note: I prefer ```=``` to construction to indicate the type returned from Get is an Address

Reply:
So I deliberately did it this way because `.Get()` has been removed from the `master` equivalent. Therefore I was planning on making my life a little easier in the future merge